modules/ting/js/ting.js

```diff
@@ -6,23 +6,8 @@
   "use strict";
 
   $(document).ready(function() {
-    $('a.js-search-overlay').on('click', function() {
-      var link = $(this);
-      if (link.attr('href').charAt(0) !== '#') {
-        // Only show overlay for non-local links.
-        Drupal.TingSearchOverlay();
-      }
-    });
-
-    // Ensure overlay on collection view links.
-    $('.ting-collection-wrapper a[href*="/ting/"]').on('click', function() {
-      if ($(this).not('[target="_blank"]').length) {
-        Drupal.TingSearchOverlay();
-      }
-    });
-
-    // Ensure overlay on object view links.
-    $('.ting-object-wrapper a[href*="/ting/"]').on('click', function() {
+    // Ensure search overlay on search links in a ting object view.
+    $('.ting-object a[href^="/search/ting/"]').on('click', function() {
```

Comment:
Before the migration to jQuery 3.x [this was also using `live()`](https://github.com/ding2/ding2/blob/7.x-4.5.0/modules/ting/js/ting.js#L18). I am not sure if it is needed. though. I cannot come up with any situations where we load content with links inside the `.ting-object` selector so perhaps this is fine.

Reply:
I can't come up with anything related to search links either. So I think it's fine. It could easily be changed from direct to delegated though.